app/src/libs/combat/util.ts

```diff
@@ -686,38 +686,53 @@ export const calcBattleResult = (battle: CompleteBattle, userId: string) => {
         result.money = moneyDelta * battle.rewardScaling + user.moneyStolen;
         // If any stats were used, distribute exp change on stats.
         // If not, then distribute equally among all stats & generals
-        let total = user.usedStats.length + user.usedGenerals.length;
+        const statsTotal = Object.values(user.usedStats).reduce((sum, value) => sum + value, 0);
+        const gensTotal = Object.values(user.usedGenerals).reduce((sum, value) => sum + value, 0);
+        let total = statsTotal + gensTotal;
         if (total === 0) {
-          user.usedStats = [
-            "ninjutsuOffence",
-            "ninjutsuDefence",
-            "genjutsuOffence",
-            "genjutsuDefence",
-            "taijutsuOffence",
-            "taijutsuDefence",
-            "bukijutsuOffence",
-            "bukijutsuDefence",
-          ];
-          user.usedGenerals = ["strength", "intelligence", "willpower", "speed"];
+          user.usedStats = {
+            ninjutsuOffence: 1,
+            genjutsuOffence: 1,
+            taijutsuOffence: 1,
+            bukijutsuOffence: 1,
+            ninjutsuDefence: 1,
+            genjutsuDefence: 1,
+            taijutsuDefence: 1,
+            bukijutsuDefence: 1,
+          }
+          user.usedGenerals = {
+            strength: 1,
+            intelligence: 1,
+            willpower: 1,
+            speed: 1,
+          };
           total = 12;
         }
         let assignedExp = 0;
-        const gain = Math.floor((experience / total) * 100) / 100;
         const stats_cap = USER_CAPS[user.rank].STATS_CAP;
         const gens_cap = USER_CAPS[user.rank].GENS_CAP;
-        user.usedStats.forEach((stat) => {
-          const value = user[stat] + gain > stats_cap ? stats_cap - user[stat] : gain;
-          result[stat] += value;
-          assignedExp += value;
+
+        const distributeStats = (
```

Comment:
I've only recently learned of the dangers and downsides of closures in javascript - I think we should take this out as a pure function instead :)